contentgrid-spring-boot-autoconfigure/src/main/java/com/contentgrid/spring/boot/autoconfigure/integration/EventsAutoConfiguration.java

```diff
@@ -0,0 +1,85 @@
+import org.springframework.hateoas.MediaTypes;
+import org.springframework.hateoas.server.mvc.TypeConstrainedMappingJackson2HttpMessageConverter;
+import org.springframework.integration.amqp.dsl.Amqp;
+import org.springframework.integration.annotation.IntegrationComponentScan;
+import org.springframework.integration.dsl.IntegrationFlow;
+import org.springframework.integration.dsl.IntegrationFlowBuilder;
+import org.springframework.integration.dsl.IntegrationFlows;
+import org.springframework.integration.dsl.Transformers;
+import org.springframework.integration.support.json.Jackson2JsonObjectMapper;
+
+import com.contentgrid.spring.integration.events.ContentGridEventPublisher;
+import com.contentgrid.spring.integration.events.ContentGridHalAssembler;
+import com.contentgrid.spring.integration.events.ContentGridMessageHandler;
+import com.contentgrid.spring.integration.events.ContentGridPublisherEventListener;
+import com.contentgrid.spring.integration.events.EntityToPersistentEntityResourceTransformer;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+@Configuration
+@IntegrationComponentScan(basePackageClasses = ContentGridEventPublisher.class)
+@EnableConfigurationProperties(EventConfigurationProperties.class)
+public class EventsAutoConfiguration {
+
+    @Bean
+    IntegrationFlow toOutboundQueueFlow(EventConfigurationProperties config,
+            ObjectProvider<ContentGridMessageHandler> handlers,
+            @Qualifier("halJacksonHttpMessageConverter") TypeConstrainedMappingJackson2HttpMessageConverter typeConstrainedMappingJackson2HttpMessageConverter,
+            ApplicationContext context) {
+
+        ObjectMapper halObjectMapper = typeConstrainedMappingJackson2HttpMessageConverter
+                .getObjectMapper();
+
+        IntegrationFlowBuilder builder = IntegrationFlows
+                .from(ContentGridEventPublisher.CONTENTGRID_EVENT_CHANNEL)
+                .transform(new EntityToPersistentEntityResourceTransformer(
+                        new ContentGridHalAssembler(context)))
+                .transform(Transformers.toJson(new Jackson2JsonObjectMapper(halObjectMapper),
+                        MediaTypes.HAL_JSON_VALUE));
+
+        handlers.stream().map(ContentGridMessageHandler::get).forEach(builder::handle);
+        return builder.get();
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    // TODO we can exclude this autoconfig by configuration property
+    ContentGridPublisherEventListener contentGridPublisherEventListener(
+            ContentGridEventPublisher contentGridEventPublisher,
+            EntityManagerFactory entityManagerFactory) {
+        return new ContentGridPublisherEventListener(contentGridEventPublisher,
+                entityManagerFactory);
+    }
+
+    @Configuration
+    @ConditionalOnProperty(value = { "spring.rabbitmq.host" })
+    static class EventsRabbitMqAutoConfiguration {
+
+        @Bean
+        @ConditionalOnProperty(value = { "contentgrid.events.queueName" })
```

Comment:
Why is the creation of this Queue been dependent on `contentgrid.events.queueName` being set? Isn't it set to a default value by default?

Reply:
yes there is a default value, but I think it might be good to have the smae approach for mq host. So, if it is not set in the config, than we do not create a queue instance and expect the default queue to be present. Not sure is it makes sense though ...

Reply:
Oh, this only creates a queue on the activemq side? It is not necessary to be able to send messages?

Reply:
First: it is rabbitmq, not activemq ;-)

If I understand the rabbitmq model correctly:
- a publisher sends to a (named) _exchange_ 
- rabbitmq is configured to route messages from an exchange to one or more queues
- consumers read from the queue

Then why are we setting up the queue on the producer-side ?

Reply:
> Oh, this only creates a queue on the activemq side? It is not necessary to be able to send messages?

yes, if the queue is not created than the message cannot be sent

Reply:
> Then why are we setting up the queue on the producer-side ?

same reason as above;) The thing is if the queue is not crated (automatically by this config, or manually) the message will not be able to be delivered. I am not aware of any other mechanism that we can use beside this one, unless rabbitmq can create a queue on the fly somehow, maybe there is some config

Reply:
> The thing is if the queue is not created (automatically by this config, or manually) the message will not be able to be delivered. 

True. But I think the queue should be created by the webhooks-server (if, and only if it is present ?)

Reply:
I want to have the message land in rabbitmq exchange - that is the end of the contract.
If there is no route/queue configured, rabbitmq can blackhole it  :shrug:

Reply:
as agreed, the starter is not responsible creating the queue. The queue should be there and exposed by the webhooks app